File: final_dataset_A/generated_code/chatgpt/python/payment_processing_endpoint_detailed-chatgpt.py
from flask import Flask, request, jsonify
from datetime import datetime

app = Flask(__name__)

@app.route("/payments/process", methods=["POST"])
def process_payment():
    # Accept payment information
    payment_request = request.get_json()

    payment_id = payment_request.get("paymentId")
    amount = payment_request.get("amount")

    # Process the payment
    status = "SUCCESS"

    # Record the transaction result
    transaction = {
        "paymentId": payment_id,
        "amount": amount,
        "status": status,
        "processedAt": datetime.now().isoformat()
    }

    # JSON response
    response = {
        "success": True,
        "message": "Payment processed successfully.",
        "transaction": transaction
    }

    return jsonify(response), 200

if __name__ == "__main__":
    app.run(debug=True)

File: final_dataset_A/generated_code/chatgpt/java/PaymentProcessingEndpointDetailed-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/payments")
public class PaymentController {

    @PostMapping("/process")
    public ResponseEntity<Map<String, Object>> processPayment(
            @RequestBody Map<String, Object> paymentRequest) {

        // Extract payment information
        String paymentId = (String) paymentRequest.get("paymentId");
        Double amount = ((Number) paymentRequest.get("amount")).doubleValue();

        // Process the payment
        String status = "SUCCESS";

        // Record the transaction result
        Map<String, Object> transaction = new LinkedHashMap<>();
        transaction.put("paymentId", paymentId);
        transaction.put("amount", amount);
        transaction.put("status", status);
        transaction.put("processedAt", LocalDateTime.now());

        // JSON response
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("success", true);
        response.put("message", "Payment processed successfully.");
        response.put("transaction", transaction);

        return ResponseEntity.ok(response);
    }
}